compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java

```diff
@@ -610,6 +618,52 @@ public void visit(BLangInvocation invocationExpr) {
         }
     }
 
+    private boolean isFieldsInitializedForSelfKeyWordExpression(BLangInvocation invocationExpr) {
+        if (invocationExpr.expr != null && invocationExpr.expr.getKind() == NodeKind.SIMPLE_VARIABLE_REF
+                && ((BLangSimpleVarRef) invocationExpr.expr).getVariableName().value.equals(Names.SELF.value)) {
+            StringBuilder uninitializedFields =
+                    getUninitializedFieldsForSelfKeyword((BObjectType) ((BLangSimpleVarRef)
+                            invocationExpr.expr).symbol.type);
+            if (uninitializedFields.length() != 0) {
+                this.dlog.error(invocationExpr.pos, DiagnosticCode.CONTAINS_UNINITIALIZED_FIELDS,
+                        uninitializedFields.toString());
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private boolean isFieldsInitializedForSelfKeyWordMethodInvocation(BLangInvocation invocationExpr) {
+
+        for (BLangExpression expr : invocationExpr.requiredArgs) {
+            if (expr.getKind() == NodeKind.SIMPLE_VARIABLE_REF && ((BLangSimpleVarRef) expr).getVariableName()
```

Comment:
Shall we extract this check to a method given that it is repeated?

Reply:
Done